### ISSUE TITLE
iOS no audio when Silent Mode is on

### ISSUE THREAD

**Issue by sahlaysta** [libretro/RetroArch#14928]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

iOS no audio when Silent Mode is on

### Expected behavior

Game audio should still work while Silent Mode is activated like most apps

### Actual behavior

Game audio does not work while Silent Mode is activated

### Steps to reproduce the bug

1. Turn on Silent Mode
2. Load game
3. Hear nothing

### Bisect Results

Today

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.14.0

### Environment information

- OS: iPhone 12 iOS 14.7.1
- Compiler: Clang/LLVM

**Comment by admonished** [comment]
Can confirm. However, headphones always work as noted in #13602

**Comment by sahlaysta** [comment]
Hey any help please. if this got fixed i would be so happy that id ejaculate.

**Comment by tobson** [comment]
I can confirm this as well.

**Comment by warmenhoven** [comment]
This is the expected behavior. When your phone is in silent mode it should be silent, including games. It's supposed to be handled by the OS.

As best I can tell, the app doesn't even know whether the switch is active or not. I didn't see an API for receiving a notification when it's toggled, and the apps I did find that notify when it's toggled are looking to see if the audio render time matches the expected duration.

**Comment by tobson** [comment]
Is it really expected behavior though? When my phone is in silent mode and I play a video on YouTube, there’s still sound. I would expect the same when playing a game.

**Comment by warmenhoven** [comment]
Yes, because YouTube is a "User-Controlled Playback App" and not a "Game App". RetroArch correctly uses `AVAudioSessionCategoryAmbient` per Apple's [Audio Guidelines By App Type](https://developer.apple.com/library/archive/documentation/Audio/Conceptual/AudioSessionProgrammingGuide/AudioGuidelinesByAppType/AudioGuidelinesByAppType.html).

**Comment by sahlaysta** [comment]
RetroArch is not on the App Store. I advocate for disregarding said norms.

**Comment by warmenhoven** [comment]
No.

Please trust that I have given this issue a great amount of thought; I do not mean this answer to be flippant or dismissive. I have spent time considering what a PR around this would look like; I did not want to give a passive-aggressive "PRs welcome" response. I actually started to create that PR, to make sure I understood all of the implications. I had a version of RetroArch running on my phone that did not obey the silent switch.

Clearly just changing from `AVAudioSessionCategoryAmbient` to `AVAudioSessionCategoryPlayback` is insufficient as then there would be complaints about RetroArch not obeying the silent switch. Further, changing it to `AVAudioSessionCategoryPlayback ` makes it so you cannot listen to music while gaming. I am sympathetic to the notion that because RetroArch is not on the App Store it does not have to abide by certain constraints, but it is still part of an ecosystem of apps and it should do its best to behave cooperatively.

So at a minimum it would need be an option. But the option is literally right there at your fingertip: the silent switch. If you want RetroArch to make noise, turn it on. I have spent time trying to understand what you're asking for and why. There are other things that the silent switch controls than just RetroArch; most essentially, I think, is that it silences notification sounds as well. That is what Focus mode is for. There is even a Gaming Focus that can turn itself on automatically when a controller is connected. I am reluctant to start complicating the codebase with platform-specific features that are reasonably accomplished through alternative platform-specific means. I am open to understanding why Game Focus does not meet your needs.

**Comment by sahlaysta** [comment]
My iPhone, iOS 14, does not have Game Focus mode. I appreciate the professional response. I see the effort you put into this issue. An added option for this would be legendary.
